rpcs3/Emu/Cell/Modules/cellRtc.cpp

```diff
@@ -488,14 +457,236 @@ error_code cellRtcFormatRfc3339LocalTime(vm::ptr<char> pszDateTime, vm::cptr<Cel
+			pszDateTime[pos], pszDateTime[pos + 1], pszDateTime[pos + 2], pszDateTime[pos + 3], pos };
+	}
+
+	const u16 year = digit(pszDateTime[pos]) * 1000 + digit(pszDateTime[pos + 1]) * 100 + digit(pszDateTime[pos + 2]) * 10 + digit(pszDateTime[pos + 3]);
+
+	pos += 4;
+
+	// Hour: " X" or " XX"
+	const u16 hour = rtcParseComponent(pszDateTime, pos, ' ', "hour");
+
+	if (hour == umax)
+	{
+		return CELL_RTC_ERROR_BAD_PARSE;
+	}
+
+	if (hour > 25) // LLE uses 25
+	{
+		return { CELL_RTC_ERROR_BAD_PARSE, "rtcParseRfc2822(): failed to parse hour: hour greater than 25" };
+	}
+
+	// Minute: ":X" or ":XX"
+	const u16 minute = rtcParseComponent(pszDateTime, pos, ':', "minute");
+
+	if (minute == umax)
+	{
+		return CELL_RTC_ERROR_BAD_PARSE;
+	}
+
+	// Second, optional: ":X" or ":XX"
+	// The string can't end with '\0' here, there must a space before it
+	u16 second = 0;
+
+	if (pszDateTime[pos] != ' ')
+	{
+		second = rtcParseComponent(pszDateTime, pos, ':', "second");
+
+		if (second == umax)
+		{
+			return CELL_RTC_ERROR_BAD_PARSE;
+		}
+	}
+	else
+	{
+		// If there are no seconds in the string, time zone requires two preceding spaces to be properly parsed
+		pos++;
+	}
+
+	// Time zone, optional, error if there is no valid time zone after the space
+	s32 time_zone = 0;
+
+	if (pszDateTime[pos] == ' ')
+	{
+		pos++;
+
+		if (pszDateTime[pos] == '+' || pszDateTime[pos] == '-')
+		{
+			// "±hhmm"
+
+			if (std::isdigit(pszDateTime[pos + 1]) &&
+				std::isdigit(pszDateTime[pos + 2]) &&
+				std::isdigit(pszDateTime[pos + 3]) &&
+				std::isdigit(pszDateTime[pos + 4]))
+			{
+				const s32 time_zone_hhmm = digit(pszDateTime[pos + 1]) * 1000 + digit(pszDateTime[pos + 2]) * 100 + digit(pszDateTime[pos + 3]) * 10 + digit(pszDateTime[pos + 4]);
+
+				time_zone = time_zone_hhmm / 100 * 60 + time_zone_hhmm % 60; // LLE uses % 60 instead of % 100
+
+				if (pszDateTime[pos] == '-')
+				{
+					time_zone = -time_zone;
+				}
+			}
+			else
+			{
+				// No error, LLE does this for some reason
+				time_zone = pszDateTime[pos] == '+' ? -1 : 1;
+			}
+		}
+		else if (pszDateTime[pos] != 'U' && pszDateTime[pos + 1] != 'T') // Case sensitive, should be || but LLE uses &&
+		{
+			// "GMT", "EST", "EDT", etc.
+
+			const u32 time_zone_idx = rtcParseName(pszDateTime, pos, TIME_ZONE_NAMES, false);
+
+			if (time_zone_idx < TIME_ZONE_NAMES.size())
+			{
+				time_zone = TIME_ZONE_VALUES[time_zone_idx] * 30;
+			}
+			else
+			{
+				// "A", "B", "C", ..., not case sensitive
+				// These are all off by one ("A" should be UTC+01:00, "B" should be UTC+02:00, etc.)
+
+				const char letter = std::toupper(pszDateTime[pos]);
+
+				if (letter >= 'A' && letter <= 'M' && letter != 'J')
+				{
+					time_zone = (letter - 'A') * 60;
+				}
+				else if (letter >= 'N' && letter <= 'Y')
+				{
+					time_zone = ('N' - letter) * 60;
+				}
+				else if (letter != 'Z')
+				{
+					return { CELL_RTC_ERROR_BAD_PARSE, "rtcParseRfc2822(): failed to parse time zone" };
+				}
```

Comment:
So I take it A-M is UTC+, N-Y is UTC-, Z is +0.
So only J and other characters are invalid ?
I kinda phased out on this last timezone part. Not gonna look at it in much more detail.

Reply:
Understandable xD
These are called "Military time zones", see here for information: https://en.wikipedia.org/wiki/Military_time_zone#Description